type error in nav/nav-logo.tsx

@genehack could you look into this [type error](https://github.com/nextstrain/nextstrain.org/actions/runs/11826364406/job/32952189773#step:5:10)? It's related to changes from https://github.com/nextstrain/nextstrain.org/pull/1032

```
Error: components/nav/nav-logo.tsx(25,12): error TS2322: Type '{ src: string; }' is not assignable to type 'string | StaticImport'.
  Type '{ src: string; }' is missing the following properties from type 'StaticImageData': height, width
```

_Originally posted by @victorlin in https://github.com/nextstrain/nextstrain.org/pull/1070#discussion_r1841231752_

I just checked locally and also don't see it. But CI on GitHub Actions is consistently showing the error on the PR. I did some [inspection](https://github.com/nextstrain/nextstrain.org/actions/runs/11923742872/job/33232726141):

- It's also using version 5.4.5 (from package-lock.json).
- The type declarations from next.js are the same as what I have locally.
- I tried removing the img type declaration with 80e36d6892b188c27e9fbaaa052d4b3c132dbcb8. `tsc` runs without error locally but the GitHub Actions run shows several `Cannot find module '…' or its corresponding type declarations.`.

Not sure how to explain the difference...

> I just checked locally and also don't see it. But CI on GitHub Actions is consistently showing the error on the PR. I did some [inspection](https://github.com/nextstrain/nextstrain.org/actions/runs/11923742872/job/33232726141):
> 
>     * It's also using version 5.4.5 (from package-lock.json).
> 
>     * The type declarations from next.js are the same as what I have locally.
> 
>     * I tried removing the img type declaration with [80e36d6](https://github.com/nextstrain/nextstrain.org/commit/80e36d6892b188c27e9fbaaa052d4b3c132dbcb8). `tsc` runs without error locally but the GitHub Actions run shows several `Cannot find module '…' or its corresponding type declarations.`.
> 
> 
> Not sure how to explain the difference...

I'll try to futz with it further…

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,6 +41,16 @@ jobs:
       - run: npm run lint:static-site
       - run: node ./scripts/check-resource-index-match.js
 
+  type-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+      - run: npm ci
+      - run: npm run type-check:ci
+
   # Build into Heroku slug so we can deploy the same build that we test.
   build:
     runs-on: ubuntu-latest

--- a/package.json
+++ b/package.json
@@ -17,6 +17,8 @@
     "lint": "npm run lint:server",
     "lint:server": "DEBUG=eslint:cli-engine npx eslint --max-warnings=0 --ext .js,.jsx .",
     "lint:static-site": "cd static-site && DEBUG=eslint:cli-engine npx eslint --max-warnings=0 --ext .js,.jsx,.ts,.tsx .",
+    "type-check": "cd static-site && tsc",
+    "type-check:ci": "npm run build && cd static-site && tsc",
     "server": "node server.js",
     "groups": "node server.js --app ./src/groupsApp.js",
     "start": "npm run server",

--- a/static-site/types.d.ts
+++ b/static-site/types.d.ts
@@ -1,23 +1,3 @@
-declare module "*.css" {
-  const value: unknown;
-  export default value;
-}
-
-declare module "*.jpg" {
-  const value: { src: string };
-  export default value;
-}
-
-declare module "*.png" {
-  const value: { src: string };
-  export default value;
-}
-
-declare module "*.svg" {
-  const value: { src: string };
-  export default value;
-}
-
 declare module "*.yaml" {
   // Actual type should be set upon import via type assertion.
   const contents: unknown;

